src/m365/flow/commands/owner/owner-add.ts

```diff
@@ -0,0 +1,170 @@
+    this.options.unshift(
+      {
+        option: '-e, --environmentName <environmentName>'
+      },
+      {
+        option: '-n, --name <name>'
+      },
+      {
+        option: '--userId [userId]'
+      },
+      {
+        option: '--userName [userName]'
+      },
+      {
+        option: '--groupId [groupId]'
+      },
+      {
+        option: '--groupName [groupName]'
+      },
+      {
+        option: '--roleName <roleName>',
+        autocomplete: FlowOwnerAddCommand.allowedRoleNames
+      },
+      {
+        option: '--asAdmin'
+      }
+    );
+  }
+
+  #initOptionSets(): void {
+    this.optionSets.push({ options: ['userId', 'userName', 'groupId', 'groupName'] });
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+        if (!validation.isValidGuid(args.options.name)) {
+          return `${args.options.name} is not a valid GUID.`;
+        }
+
+        if (args.options.userId && !validation.isValidGuid(args.options.userId)) {
+          return `${args.options.userId} is not a valid GUID.`;
+        }
+
+        if (args.options.userName && !validation.isValidUserPrincipalName(args.options.userName)) {
+          return `${args.options.userName} is not a valid user name.`;
+        }
+
+        if (args.options.groupId && !validation.isValidGuid(args.options.groupId)) {
+          return `${args.options.groupId} is not a valid GUID.`;
+        }
+
+        if (FlowOwnerAddCommand.allowedRoleNames.indexOf(args.options.roleName) < 0) {
+          return `${args.options.roleName} is not a valid role name. Valid role names are ${FlowOwnerAddCommand.allowedRoleNames.join(', ')}`;
+        }
+
+        return true;
+      }
+    );
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    try {
+      if (this.verbose) {
+        logger.logToStderr(`Assigning permissions to a Power Automate flow in environment ${args.options.environmentName}`);
+      }
+
+      let id = '';
+      if (args.options.userId) {
+        id = args.options.userId;
+      }
+      else if (args.options.userName) {
+        id = await aadUser.getUserIdByUpn(args.options.userName);
+      }
+      else if (args.options.groupId) {
+        id = args.options.groupId;
+      }
+      else {
+        const resp = await aadGroup.getGroupByDisplayName(args.options.groupName!);
+        id = resp.id!;
+      }
+
+      const requestOptions: CliRequestOptions = {
+        url: `${this.resource}providers/Microsoft.ProcessSimple/${args.options.asAdmin ? 'scopes/admin/' : ''}environments/${formatting.encodeQueryParameter(args.options.environmentName)}/flows/${formatting.encodeQueryParameter(args.options.name)}/modifyPermissions?api-version=2016-11-01`,
+        headers: {
+          accept: 'application/json'
+        },
+        data: {
+          "put": [
+            {
+              "properties": {
+                "principal": {
+                  "id": id
```

Comment:
While retrieving the owners after setting them, seems like the "type" value is empty.

```json
{
"properties": {
        "permissionType": "Principal",
        "principal": {
          "id": "2551fd4e-9072-43f6-971d-3cc9c7e9b0b2",
          "type": "NotSpecified"
        }
      }
}
```

I suggest that we set the `type` value in our request with either `User` or `Group`.